sys/posix/sockets/posix_sockets.c

```diff
@@ -82,6 +96,8 @@ typedef struct {
     int type;
     int protocol;
     bool bound;
+    uint8_t recv_buf[RECV_BUF_SIZE];
+    ringbuffer_t peek_rb;
```

Comment:
Given that this adds _**1.5 KB** per allocatable socket_ I would rather like to see this feature to be optional. For instance, by introducing a pseudo-module `posix_sockets_msg_peek`.